libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1208,7 +1205,7 @@ void CompilerUtils::convertType(
 			}
 			break;
 		case DataLocation::CallData:
-			solAssert(false, "Invalid type conversion target location CallData.");
+			// nothing to do
```

Comment:
Can you add an assert that the types are the same? Does that make sense?

Reply:
Hm, we don't do it for the other locations - or did I just not find the check?

Reply:
Yeah, seems like a good idea. Let's see if tests pass with it.

Reply:
> Hm, we don't do it for the other locations - or did I just not find the check?

Yeah, you were right. Not it's fixed.